pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -146,16 +142,6 @@ func getClassInfos() map[gateway.GatewayController]classInfo {
 		},
 	}
 
-	if features.MultiNetworkGatewayAPI {
-		m[constants.UnmanagedGatewayController] = classInfo{
-			// This represents a gateway that our control plane cannot discover directly via the API server.
-			// We shouldn't generate Istio resources for it. We aren't programming this gateway.
-			controller:             constants.UnmanagedGatewayController,
-			description:            "Remote to this cluster. Does not deploy or affect configuration.",
-			disableRouteGeneration: true,
-			addressType:            gateway.HostnameAddressType,
```

Comment:
cc @stevenctl

Reply:
Yes，https://github.com/istio/istio/blob/c71b83f44e2c82e0c62c3a54f1e21c017cef0a10/pilot/pkg/serviceregistry/kube/controller/network.go#L350
  we get the network address from gateway with `istio-remote` class name. 

But this classinfo is used for convert kube gateway to istio gateway resources. So a gateway with `istio-remote` class name should be skipped here  https://github.com/istio/istio/blob/c71b83f44e2c82e0c62c3a54f1e21c017cef0a10/pilot/pkg/config/kube/gateway/conversion.go#L1986-L1988

Reply:
classInfos is also used for GatewayClass creation

Reply:
... why? There is supposed to be a GatewayClass for every class we support. We support istio-remote, so there should be a GWC.

Reply:
GatewayClass is something like ingressclass, which is not a must.

Reply:
I still dont see why we should remove it. Even if it's not *strictly* required it's still useful?

Reply:
We donot check istio-remote gateway class, nor using it to deploy a gateway.

Reply:
Yeah GatewayClass is basically documentation. We don't use it in the "istio" class either. 

I'm pretty sure it's required in conformance as well